code/controllers/subsystem/timer.dm

```diff
@@ -259,11 +259,11 @@ SUBSYSTEM_DEF(timer)
 			var/datum/timedevent/Last = ctts[cttl]
 			if(Last.timeToRun >= timeToRun)
 				ctts += src
-			else if(cttl > 1)
-				for(var/I in cttl to 1)
-					var/datum/timedevent/E = ctts[I]
+			else
+				for(var/i=cttl,i>=1,i--)
```

Comment:
`for(var/i in cttl to 1 step -1)` will fix this in the same way while maintaining the performance benefit of using ranges